### DISCUSSION TITLE
Log Panel ability to narrow down fields

### DISCUSSION THREAD

**Discussion by x3nb63** [grafana/grafana#45720] [Dashboards, Panels & Visualizations]
The detected fields make the Log panel hard to read in my case, see screenshot below. The json format totally confuses, esp. since these logs are multi-line printouts, ie there is a table of numbers in the middle. All I would want to see is the content of the `log` field, ideally with a fixed-width font.

At the same time the other fields are nice to have, when expanding a line to see all its details such as that it came via `stdout` - some colorization based on the value of another field would be really cool.

I read #35044 and tried to use the Table panel + Transformations but failed to take apart the line values by regex or somehow else.

![log-panel-with-json-fields](https://user-images.githubusercontent.com/25959223/155120368-b84e0049-966f-49d1-84b3-381ff1e73bc4.PNG)

This far its nice to have the logs show up in dashboards, but they are harder to read then by looking directly into the files (which non-cluster-admins cant do).

**Comment by x3nb63** [comment]
having that in practical use for  a week I spot another improvement:

- the ability to download the shown log lines as text file  and/or copy to clipboard

for example to paste into a ticket or as input to further processing ...

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
